packages/web/components/Quest/QuestDetailsHeader.tsx

```diff
@@ -21,12 +43,74 @@ export const QuestDetailsHeader: React.FC<Props> = ({ quest }) => {
   const {
     repetition,
     cooldown,
-    // createdAt,
-    // externalLink: link,
     title,
     status,
+    quest_skills: questSkills,
+    quest_roles: questRoles,
   } = quest;
 
+  // Archive quest modal
+  const cancelRef = useRef<HTMLButtonElement>(null);
+  const archiveButtonRef = useRef<HTMLButtonElement>(null); // focus returns here when modal is closed
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  const toast = useToast();
+  const router = useRouter();
+  const [, updateQuest] = useUpdateQuestMutation(); // Don't need to use updateQuestResult that's returned from useUpdateQuestMutation e.g const [updateQuestResult, updateQuest] = ...
+
+  // Original skills and roles are needed (required field) for the updateQuest mutation
+  // because the updateQuest mutation deletes existing skills/roles and then adds in those
+  // that were submitted via the form (either the existing skills/roles or changes to them)
+  // If we keep the original skills/roles in, the quest can still be unarchived if needed
+
+  // Roles objects
+  const rolesObjects = questRoles.map((r) => ({
+    questId: quest.id,
+    role: r.PlayerRole.role,
+  }));
+
+  // Skills objects
+  const skillsObjects = questSkills.map((s) => ({
+    questId: quest.id,
+    skillId: s.skill.id,
+  }));
+
+  const setQuestAsArchived = () => {
```

Comment:
Why not `useCallback` and `async` ?

Reply:
I've got everything in the dependencies for the `useCallback`

`}, [quest.id, questSkills, questRoles, onClose, router, toast, updateQuest]);`

I wanted to just have `quest.id`, `questSkills`, `questRoles` because I thought those were the relevant ones, but that gives a linting error.

What's better for the dependencies - all in, or none in?

Reply:
yes, the lint rule will tell you which variables are being referenced in the hook and should are dependencies. With no dependencies the hook will be reinstantiated on every render, which kind of defeats the point.